Sources/Network/JSON.swift

```diff
@@ -61,37 +61,55 @@ public enum JSON {
         return try parse(from: data)
     }
 
-    /// Parse an attribute of type `T` with the given attribute on the given JSON dictionary, and optionally validate 
+    /// Parse an attribute of type `T` with the given key on the given JSON dictionary, and optionally validate
     /// the value using a `where` predicate closure. Additionally, an optional `parseAPIError` closure can be supplied
     /// so that if a parsing step fails an attempt is made to extract a domain specific error and throw it.
     ///
     /// - Parameters:
-    ///   - attribute: The JSON attribute key to parse.
+    ///   - type: The type of the attribute to parse.
+    ///   - key: The JSON attribute key to parse.
     ///   - json: The JSON dictionary.
     ///   - predicate: The validation predicate.
     ///   - parseAPIError: The API error parsing closure.
-    /// - Returns: The attribute of type `T` asociated to the given key.
+    /// - Returns: The value of type `T` associated to the given attribute's key.
     /// - Throws: An error of type `JSON.Error`, or a domain specific `Swift.Error` produced by `parseAPIError`.
-    public static func parseAttribute<T>(_ key: JSON.AttributeKey,
+    public static func parseAttribute<T>(_ type: T.Type,
```

Comment:
Well, I honestly don't think it's necessary to think about backwards compatibility at this point of the framework's life. 

So far we have consciously avoided versioning the framework precisely to give us complete flexibility over the API's and not have to think about those types of issues until we feel it has reached a certain level of maturity.

In the future I believe we intend to follow semantic versioning and then we will have to concern ourselves with these issues whenever we "break" or deprecate an API.

As for taking advantage of the type inference, do you really think it's worth it?